src/battle_ai_main.c

```diff
@@ -4421,7 +4442,10 @@ static s16 AI_CheckViability(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
             score++;
         break;
     case EFFECT_REFRESH:
-        if (gBattleMons[battlerAtk].status1 & STATUS1_ANY)
+    case EFFECT_TAKE_HEART:
+        if (gBattleMons[battlerAtk].status1 & STATUS1_ANY
+         || BattlerStatCanRise(battlerAtk, AI_DATA->abilities[battlerAtk], STAT_SPATK)
```

Comment:
Now this is applying Take Heart's logic to Refresh even though Refresh does not raise stats.

Reply:
Yes.

Since Refresh does not raise any stats, those checks will neve really be reached by it.

I could split them, I just took the lazy route because it doesn't matter functionally speaking 😂

Reply:
Functionally speaking it does matter. If Refresh falls through to Take Heart's logic it would tell the AI to use Refresh if it's Sp. Atk or Sp. Def can rise (which is wrong).

But thank you for fixing it.